bot/exts/utils/reminders.py

```diff
@@ -30,6 +32,133 @@
 ReminderMention = t.Union[UnambiguousUser, discord.Role]
 
 
+class SnoozeSelectView(discord.ui.View):
+    """The reminder's select dropdown UI View."""
+
+    SNOOZE_DURATIONS: dict[str, t.Optional[int]] = {
+        # Mapping of `duration string: num_seconds_in_duration`.
+        #
+        # `num_seconds_in_duration` is `None` for durations >= 1 month
+        # since the amount of seconds depends on the date of lookup
+        # (i.e. seconds from 1st jan to 1st feb is different than from 1st feb to 1st march etc).
+        "1 minute": 60,
+        "5 minutes": 5 * 60,
+        "15 minutes": 15 * 60,
+        "30 minutes": 30 * 60,
+        "1 hour": 60 * 60,
+        "3 hours": 3 * 60 * 60,
+        "6 hours": 6 * 60 * 60,
+        "1 day": 24 * 60 * 60,
+        "1 week": 7 * 24 * 60 * 60,
+        "1 month": None,
+        "3 months": None,
+        "6 months": None,
+        "1 year": None
+    }
+
+    def __init__(self, bot: Bot, reminder_id: int, reminder_author_id: int):
+        super().__init__(timeout=30)
+        self.new_expiry: t.Optional[datetime] = None
+        self.reminder_was_snoozed = False
+
+        self.bot = bot
+        self.reminder_id = reminder_id
+        self.interaction_owner_id = reminder_author_id
+
+        self.dropdown: discord.ui.Select = self.children[0]
+        for duration in self.SNOOZE_DURATIONS:
+            self.dropdown.add_option(label=duration)
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        """Ensure that the user clicking the button is the member who invoked the command."""
+        if interaction.user.id != self.interaction_owner_id:
+            await interaction.response.send_message(":x: This is not your reminder to snooze!", ephemeral=True)
+            return False
+        return True
+
+    @discord.ui.select(placeholder="Select the snooze duration")
+    async def select_snooze_duration(self, _: discord.ui.Select, interaction: discord.Interaction) -> None:
+        """Drop down menu which contains a list of snooze durations one can choose."""
+        selected_duration = interaction.data["values"][0]
+        current_datetime = datetime.now()
```

Comment:
I believe this is naive, you should most likely supply it with a tomezone.

Reply:
Is there a point for this? We aren't passing this to dpy methods at all.

Reply:
Even if we don't pass it discord.py, if we make it a habit of always using aware datetime objects everywhere then we'll never have issues with aware/naive datetimes anywhere. I think it's a good habit to maintain.

Reply:
Yep, I've fixed this by doing `datetime.now(timezone.utc)`, as per [the docs'](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) recommended way.